src/NServiceBus.Testing.Tests/HandlerTests.cs

```diff
@@ -194,6 +202,14 @@ public void ShouldPassExpectPublishWhenPublishing()
         public void ShouldFailExpectNotPublishWhenPublishing()
         {
             Assert.Throws<ExpectationException>(() => Test.Handler<PublishingHandler<Publish1>>()
+                .ExpectNotPublish<Publish1>()
+                .OnMessage<TestMessage>());
+        }
+
+        [Test]
+        public void ShouldFailExpectNotPublishWithCheckWhenPublishing()
+        {
+            Assert.Throws<ExpectationException>(() => Test.Handler<PublishingHandler<Publish1>>()
                 .ExpectNotPublish<Publish1>(m => true)
```

Comment:
yeah would make more sense. I'm more afraid of the effort and whether this is worth it?